flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/compiler/NFACompiler.java

```diff
@@ -158,9 +158,10 @@ public static boolean canProduceEmptyMatches(final Pattern<?, ?> pattern) {
          */
         void compileFactory() {
             if (currentPattern.getQuantifier().getConsumingStrategy()
-                    == Quantifier.ConsumingStrategy.NOT_FOLLOW) {
+                            == Quantifier.ConsumingStrategy.NOT_FOLLOW
```

Comment:
Currently, the windowTime is the minimum of all window times and so it may happen that the window time is defined in the other Pattern. What about moving this check to the end of this method?